### DISCUSSION TITLE
Keyboard/mouse not recognized - DOS to Win3.1 install on iPadOS

### DISCUSSION THREAD

**Discussion by yllekz** [utmapp/UTM#6488] [UTM Help (iOS)]
Hi folks, I know the official App Store iOS/iPadOS app is in its early days, but I ran into an issue earlier and wasn't sure if it was a bug or intentional.

I successfully installed MS-DOS 6.22 and then wanted to install Windows 3.1 on top of it. Things went smoothly until I got to the "Windows Setup" portion which requires one to input the information via the GUI with keyboard/mouse input. The problem is, input is not recognized at this point in the setup.  It was fine before then. Do I have to load some kind of extra driver?

**Comment by extratone** [comment]
Apologies for not bringing more details atm, but... after a good portion of today spent exploring UTM SE for the first time on my `iPad13,9`, I just want to make sure to note that "classic" (read: neglected for years, encountered multiple times a day with any substantial use) bugs/inconsistencies in iOS/iPadOS' hardware keyboard support was almost certainly causing most of the (many) I/O issues I had. 

(Just wanna make sure it doesn't get misattributed and waste anybody's time.)

**Comment by micturkey** [comment]
Disable USB support in INPUT setting of VM.
I encountered this issue both in setting up Win3.2 and Win98, and by disabling USB it works.

**Reply by TotallyNotTech** [reply]
Yep that works! 🥳
